plugins/inputs/ecs/client.go

```diff
@@ -0,0 +1,124 @@
+package ecs
+
+import (
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"time"
+
+	"github.com/docker/docker/api/types"
+)
+
+var (
+	ecsMetadataPath, _  = url.Parse("/v2/metadata")
+	ecsMetaStatsPath, _ = url.Parse("/v2/stats")
+)
+
+// Client is the ECS client contract
+type Client interface {
+	Task() (*Task, error)
+	ContainerStats() (map[string]types.StatsJSON, error)
+}
+
+type httpClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+// NewEnvClient configures a new Client from the env
+func NewEnvClient() (*EcsClient, error) {
```

Comment:
You can remove this function, envcfg option and docs

https://github.com/influxdata/telegraf/pull/5121#discussion_r281870269